include/boost/geometry/algorithms/detail/overlay/get_distance_measure.hpp

```diff
@@ -0,0 +1,188 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_OVERLAY_GET_DISTANCE_MEASURE_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_OVERLAY_GET_DISTANCE_MEASURE_HPP
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/core/coordinate_system.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+#include <boost/geometry/util/select_coordinate_type.hpp>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail
+{
+
+template <typename T>
+struct distance_measure
+{
+    T value;
+
+    distance_measure()
+        : value(T())
+    {}
+
+    bool is_small() const { return true; }
+    bool is_zero() const { return true; }
+    bool is_positive() const { return false; }
+};
+
+template <typename T>
+struct distance_measure_floating
+{
+    T measure;
+
+    distance_measure_floating()
+        : measure(T())
+    {}
+
+    // Returns true if the distance measure is small.
+    // This is an arbitrary boundary, to enable some behaviour
+    // (for example include or exclude turns), which are checked later
+    // with other conditions.
+    bool is_small() const { return std::fabs(measure) < 1.0e-3; }
+
+    // Returns true if the distance measure is absolutely zero
+    bool is_zero() const { return measure == 0.0; }
+
+    // Returns true if the distance measure is positive. Distance measure
+    // algorithm should return positive value if it is located on the left side.
+    bool is_positive() const { return measure > 0.0; }
+};
+
+template <>
+struct distance_measure<long double>
+    : public distance_measure_floating<long double> {};
+
+template <>
+struct distance_measure<double>
+    : public distance_measure_floating<double> {};
+
+template <>
+struct distance_measure<float>
+    : public distance_measure_floating<float> {};
+
+} // detail
+
+
+namespace detail_dispatch
+{
+
+template <typename CalculationType, typename Tag>
+struct get_distance_measure
+        : not_implemented<Tag>
+{};
+
+template <typename CalculationType>
+struct get_distance_measure<CalculationType, cartesian_tag>
+{
+    typedef detail::distance_measure<CalculationType> result_type;
+
+    template <typename SegmentPoint, typename Point>
+    static result_type apply(SegmentPoint const& p1, SegmentPoint const& p2,
+                             Point const& p)
+    {
+        typedef CalculationType ct;
+
+        // Construct a line in general form (ax + by + c = 0),
```

Comment:
if this is only for cartesian shouldn't it be in strategies?

Reply:
I did not make it a real strategy, invokeable by users - it is a detail, so for now I kept it here. Unless you object.

Reply:
Ok, but maybe add a comment or TODO that this should be moved to strategies in the future?

Reply:
> I did not make it a real strategy, invokeable by users

It's not a matter of invoking by the user since it should still be derived from higher-level strategy like intersection strategy in this case.

I guess we could keep it here (at least for now) but could you at least take the coordinate system from strategy instead of geometry? You could add `typedef ... cs_tag` member type to intersection strategies for this purpose.

Actually we could do this with all strategies in order to generate CS-specific parts of the algorithm if needed instead of defining strategies within strategies.

Reply:
But there is another thing I don't understand. AFAIU this is a utility to make sides calculation more robust, correct? consistent with distance calculation? If that's the case why is this not a part of side strategy? Or does it have a different purpose?

Reply:
Working at this:

> I guess we could keep it here (at least for now) but could you at least take the coordinate system from strategy instead of geometry? You could add typedef ... cs_tag member type to intersection strategies for this purpose.

But I'm a puzzled - what value does it add? I assume you find it more configurable because that tag comes from intersection strategy, and not from geometry, and they can be passed independently.

We now use the basic intersection strategy as an umbrella strategy to access all other strategies. So yes, I agree that a cs_tag could be added there. But I think it should be splitted again - one basic intersection strategy (as we had before) and a separate umbrella strategy returning you all the other strategies (area, point-in-geometry, envelope, etc) which all have nothing to do with intersection strategy itself (that, as far as I can see, needs only a side strategy as a dependency).

Anyway, that is a different subject. To make it just a bit more consistent with current implementations I will add that tag - but I'm really hesitant to add an accessor there again for this new strategy.